src/resolve/resolved-mdns.c

```diff
@@ -406,6 +433,22 @@ static int on_mdns_packet(sd_event_source *s, int fd, uint32_t revents, void *us
                                 log_debug("Got a goodbye packet");
                                 /* See the section 10.1 of RFC6762 */
                                 rr->ttl = 1;
+
+                                /* Look at the cache 1 second later and remove stale entries.
+                                 * This is particularly useful to keep service browsers updated on service removal,
+                                 * as there are no other reliable triggers to propogate that info. */
+                                if (!scope->mdns_goodbye_event_source) {
+                                        r = sd_event_add_time_relative(
```

Comment:
Still I cannot follow why we need to actively clear the outdated cache entries...

Reply:
In the context of continuous browsing (#22532), it would be useful to actively clear the outdated cache entries and notify the browser client(s).
To accomplish that, we would eventually add a call to mdns_notify_browsers_unsolicited_updates()
from mdns_goodbye_callback().